extensions/kubernetes-client/deployment-internal/src/main/java/io/quarkus/kubernetes/client/deployment/NoQuarkusTestKubernetesClient.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.kubernetes.client.deployment;
+
+import java.util.Arrays;
+import java.util.function.BooleanSupplier;
+
+class NoQuarkusTestKubernetesClient implements BooleanSupplier {
```

Comment:
I don't see what is the purpose of this logic, can you please clarify?

Reply:
This refer to the first point in description. `io.quarkus.test.kubernetes.client` package contains all quarkus mock server test class used to actually mock kubernetes. This is the package that I think most people use to write unit test. It is not compatible with dev services because they barely do the same thing to configure a KubernetesClient to connect to the test server. Dev services wins because quarkus properties override the system props properties, but it can't detect it because systemProps are set latter by a test resource class. Thus, all existing unit test would fail if we don't disable dev services when the mock package is used.

Reply:
I think that we need a couple of things here:
- a more descriptive name
- a comment that explain(s) why?
- documentation
- (optionally) create a feature requrest to allow for multiple clients with qualifiers.

Reply:
Added documentation and guide. First time doing this.